src/AutoRest.CSharp/Common/Input/InputTypes/InputPrimitiveType.cs

```diff
@@ -3,37 +3,66 @@
 
 namespace AutoRest.CSharp.Common.Input;
 
-internal record InputPrimitiveType(InputPrimitiveTypeKind Kind) : InputType(Kind.ToString())
+internal record InputPrimitiveType(InputPrimitiveTypeKind Kind, string Name, string CrossLanguageDefinitionId) : InputType(Name)
 {
-    internal InputPrimitiveType(InputPrimitiveTypeKind kind, string? encode) : this(kind)
+    #region Scalars defined in typespec-azure-core
+    internal const string UuidId = "Azure.Core.uuid";
+    internal const string IPv4AddressId = "Azure.Core.ipV4Address";
+    internal const string IPv6AddressId = "Azure.Core.ipV6Address";
+    internal const string ETagId = "Azure.Core.eTag";
+    internal const string AzureLocationId = "Azure.Core.azureLocation";
+    internal const string ArmIdId = "Azure.Core.armResourceIdentifier";
+    #endregion
+
+    #region Types we supported but not yet added to typespec-azure-core
+    internal const string CharId = "Azure.Core.char";
+    internal const string ContentTypeId = "Azure.Core.contentType";
+    internal const string ResourceTypeId = "Azure.Core.resourceType";
+
+    internal const string ObjectId = "Obsolete.object";
```

Comment:
It seems `ObjectId` and `IPAddressId` will never be added to typespec-azure-core, shall we put them to another region and add some comments explaining they are just markers of legacy types for TypeFacotry?

Reply:
sure